src/main/java/com/launchdarkly/client/FeatureFlag.java

```diff
@@ -0,0 +1,242 @@
+package com.launchdarkly.client;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.reflect.TypeToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.util.*;
+
+public class FeatureFlag {
+  private final static Logger logger = LoggerFactory.getLogger(FeatureFlag.class);
+
+  private static final Gson gson = new Gson();
+  private static final Type mapType = new TypeToken<Map<String, FeatureFlag>>() {
+  }.getType();
+
+  private String key;
+  private int version;
+  private boolean on;
+  private List<Prerequisite> prerequisites;
+  private String salt;
+  private String sel;
+  private List<Target> targets;
+  private List<Rule> rules;
+  private Rule fallthrough;
+  private Integer offVariation; //optional
+  private List<JsonElement> variations;
+  private boolean deleted;
+
+  public static FeatureFlag fromJson(String json) {
+    return gson.fromJson(json, FeatureFlag.class);
+  }
+
+  public static Map<String, FeatureFlag> fromJsonMap(String json) {
+    return gson.fromJson(json, mapType);
+  }
+
+  public FeatureFlag(String key, int version, boolean on, List<Prerequisite> prerequisites, String salt, String sel, List<Target> targets, List<Rule> rules, Rule fallthrough, Integer offVariation, List<JsonElement> variations, boolean deleted) {
+    this.key = key;
+    this.version = version;
+    this.on = on;
+    this.prerequisites = prerequisites;
+    this.salt = salt;
+    this.sel = sel;
+    this.targets = targets;
+    this.rules = rules;
+    this.fallthrough = fallthrough;
+    this.offVariation = offVariation;
+    this.variations = variations;
+    this.deleted = deleted;
+  }
+
+  JsonElement getOffVariation() {
+    if (offVariation != null && offVariation < variations.size()) {
+      return variations.get(offVariation);
+    }
+    return null;
+  }
+
+  EvalResult evaluate(LDUser user, FeatureStore featureStore) {
+    if (user == null || user.getKey() == null) {
+      return null;
+    }
+    Set<FeatureRequestEvent> prereqEvents = new HashSet<>();
```

Comment:
Using a set here may not make sense, as the `FeatureRequestEvent`s don't have reasonable equals / hashcode implementations.